### ISSUE TITLE
Drop an index not working in Chrome

### ISSUE THREAD

**Issue by qoregexp** [dfahlander/Dexie.js#981]
For performance reasons, I want to get rid of an index on one table.  Everything seems to work OK if I just delete the index in my code, but I thought I should probably go to a new version.  The documentation says the index will automatically be dropped if not specified in the new version, but this isn't happening in Chrome.  It does appear to have dropped the index in Firefox (this is on Windows 7).  I closed Chrome out and restarted it in case the index might go away but it didn't change anything.  I get the performance improvement with the index not being maintained, but it's big table and it would be nice to have the index go away.

**Comment by swedenborg** [comment]
i think the new version do not change the previous version index as it is versioned ;-)

i found this code sample that might help ...

```
//
// !!Please open your console in devtools to see any action!!
//

async function changeSchema(db, schemaChanges) {
	db.close();
	const newDb = new Dexie(db.name);

	newDb.on('blocked', () => false); // Silence console warning of blocked event.
	// Workaround: If DB is empty from tables, it needs to be recreated
	if (db.tables.length === 0) {
		await db.delete();
		newDb.version(1).stores(schemaChanges);
		return await newDb.open();
	}

	// Extract current schema in dexie format:
	const currentSchema = db.tables.reduce((result, { name, schema }) => {
		result[name] = [ schema.primKey.src, ...schema.indexes.map((idx) => idx.src) ].join(',');
		return result;
	}, {});

	console.log('Version: ' + db.verno);
	console.log('Current Schema: ', currentSchema);

	// Tell Dexie about current schema:
	newDb.version(db.verno).stores(currentSchema);
	// Tell Dexie about next schema:
	newDb.version(db.verno + 1).stores(schemaChanges);
	// Upgrade it:
	return await newDb.open();
}

// Open database dynamically:
async function playAround() {
	let db = new Dexie('FriendsDatabase2');
	if (!await Dexie.exists(db.name)) {
		console.log('Db does not exist');
		db.version(1).stores({});
	}
	await db.open();
	console.log('Could open DB');

	// Add a table with some indexes:
	db = await changeSchema(db, {
		friends: 'id, name',
	});
	console.log('Could enforce friends table with id and name');

	// Add another index in the friends table
	db = await changeSchema(db, {
		friends: 'id, name, age',
	});
	console.log('Could add the age index');

	// Remove the age index again:
	db = await changeSchema(db, {
		friends: 'id, name',
	});
	console.log('Could remove age index');

	// Remove the friends table
	db = await changeSchema(db, {
		friends: null,
	});
	console.log('Could delete friends table');
}

playAround().catch((err) => console.error(err));
```

**Comment by qoregexp** [comment]
I think this is doing basically what I'm doing -- declaring a new version without as many indices as the previous.  The database version gets upped (I can see that in the dev console) but the index is not removed.  My case involves 2 compound keys -- I am going from 
		chart: '[patno+row+col], [patno+done]'
to
		chart: '[patno+row+col]'
and that may make a difference.

**Comment by dfahlander** [comment]
@qoregexp How, in what way are you seeing that the index is not going away? Does it still show up in the Application tab in chrome devtools under the indexeddb / yourTable / yourIndex? 

Also wondering which version of dexie you are using, as there's a recent PR (#959) present in the latest versions (3.0.0-rc.3 and 3.0.0-rc.4) that changes the behaviour of upgrades. Just to make sure no bug has come into dexie. It would be nice if you could console log `diff` in schema-helpers.tsx, line 98 before going through changes, and then also on line 112, where it is supposed to delete indexes, if you suspect a bug in dexie. If you are using a build system like webpack or rollup, the easiest way to alter the source is to clone this git repo, build it and do npm link to replace dexie with your own version (as described [here](https://github.com/dfahlander/Dexie.js#contributing)).

**Comment by qoregexp** [comment]
@dfahlander Yes, it still shows the second index on the side under the table name, and the index values are all still there.  
![Dexie example](https://user-images.githubusercontent.com/7798324/75123277-fb907e80-5673-11ea-84ea-c913c4a6fa2c.jpg)

I had been experimenting with 3.00-rc.4 but I went back to the released version (@latest) before I added the new schema version.  So this is the result of having a version 3 db with 2 indexes on the chart table, then upgrading to version 4 with 1 index.  

However, if I delete the database, open it as a new empty version 3 database, then upgrade to version 4, although the patno+done key still shows on the left, it doesn't get populated when I add data to the db.  And I can tell by the performance that even in the case data was added to the db as version 3 and the second index has values, that once it converts to version 4, the index is not being maintained.

I don't necessarily think this is a bug in Dexie -- maybe the Chrome implementation of indexedDB.  I   would be happy to do the debugging things you ask but I have never used npm and it may take me a while to figure it out.  I could easily give you a minimal test case though.  I should have done that anyway, sorry.  I will put something together and get back to you.

**Comment by dfahlander** [comment]
Thanks a lot for the detailed explanation. I assume this is a chromium issue as you are seeing a difference in how the index is stopped being maintained after the upgrade when you inspect it using chrome devtools. If you'd have the time to repro and file this issue in chromium issue tracker, please do. If not, let us know and hopefully someone else could pick this up and file it.

1. Create a repro that creates and upgrades a DB.
    * Verify that IDBObjectStore.deleteIndex() has indeed been called in (to eliminate it being a bug in Dexie)
    * If not using npm you could edit the monolit [dexie.js](https://unpkg.com/dexie@3.0.0-rc.4/dist/dexie.js) - search for "deleteIndex" and console.log or similar when call is made.
2. Go file the issue at https://bugs.chromium.org/p/chromium/issues and provide a link to the repro and to this original dexie ticket

**Comment by qoregexp** [comment]
I wrote a simple test case to give you and to my infinite shame, it works correctly.  So I am trying to narrow down WHAT about my more complex actual code causes the index not to be dropped.  I will keep working on it and get back to you when I know more.

**Comment by dfahlander** [comment]
No problems at all! It might still be something in a certain condition that made the browser keep the index. You mentioned the database was very large. Maybe that must also be reproduced to trigger what you have seen. Maybe chromium was deleting the index very lazily for large DBs for example. If so, I would assume the index would go away eventually but not as eagerly as you probably desired.

**Comment by qoregexp** [comment]
There is no problem at all, except that I apparently was extremely confused.  Printing out the secondary indexes from the schema shows that the second index is deleted, and I thought that the problem was just that Chrome wasn't updating its display of what was in the db, but even that was not true.  :(  I'm sorry.  (But my testing of the performance issue that led me to drop the second index was less boneheaded -- do you still check stackoverflow sometimes or should I open a new issue here?)

**Comment by dfahlander** [comment]
I subscribe to stackoverflow questions tagged dexie
